src/Promitor.Integrations.AzureMonitor/Extensions/MetricResultExtension.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Text.RegularExpressions;
+using Azure.Monitor.Query.Models;
+
+namespace Promitor.Integrations.AzureMonitor.Extensions
+{
+    public static class MetricResultExtension 
+    {   
+        // hacky to to get resource ID since it's not available directly through the SDK model
+        static string resourceIdPattern = @"^([\/]?subscriptions\/[^\/]+\/resourceGroups\/[^\/]+\/providers\/[^\/]+\/[^\/]+\/[^\/]+)";     
+        static Regex resourceIdRegex = new Regex(resourceIdPattern, RegexOptions.Compiled);
+        
+        public static string ParseResourceIdFromResultId(this MetricResult metricResult) 
+        {
```

Comment:
Is this still required?

Reply:
Yes unfortunately, because the SDK does not have a `ResourceID` field that's part of the JSON response. I've added comment in this class to explain the situation